frontend/src/GameRouter.tsx

```diff
@@ -1,29 +1,101 @@
+import createPersistedState from "use-persisted-state";
 import { BasicPage, NicknamePage, LobbyPage } from "./pages";
+import socket from "./socket";
+import { PlayersContext } from "./ContextProviders/PlayersContextProvider";
+
+export type Settings = {
+  roundLimit?: number;
+  maxPlayers?: number;
+};
 
 type PathParams = {
   gameCode: string;
 };
 
+const memoryHistory = createMemoryHistory();
+const usePlayerIdState = createPersistedState("playerId");
+const useTokenState = createPersistedState("token");
+
 const GameRouter = () => {
   const { gameCode } = useParams<PathParams>();
+  const [, setPlayerId] = usePlayerIdState("");
+  const [, setToken] = useTokenState("");
+  const { setHost, initialisePlayers, addPlayer, removePlayer } = useContext(
+    PlayersContext
+  );
+  const [settings, setSettings] = useState<Settings>();
+
+  const handleNavigate = useCallback(memoryHistory.push, [memoryHistory.push]);
+  const handleHost = useCallback(setHost, [setHost]);
+  const handlePlayersInitial = useCallback(initialisePlayers, [
+    initialisePlayers,
+  ]);
+  const handlePlayersAdd = useCallback(addPlayer, [addPlayer]);
+  const handlePlayersRemove = useCallback(removePlayer, [removePlayer]);
+  const handleSettingsInitial = useCallback(setSettings, [setSettings]);
+  const handleSettingsUpdate = useCallback(
+    ({ setting, value }: { setting: string; value: number }) => {
+      let key;
+      switch (setting) {
+        case "MAX_PLAYERS":
+          key = "maxPlayers";
+          break;
+        case "ROUND_LIMIT":
+          key = "roundLimit";
+          break;
+        default:
+          key = null;
+      }
+      if (key !== null) {
+        const newSettings = {
+          ...settings,
+          [key]: value,
+        };
+        setSettings(newSettings);
+      }
+    },
+    [setSettings]
+  );
+  const handleConnectError = useCallback(() => {
+    setPlayerId("");
+    setToken("");
+  }, [setPlayerId, setToken]);
+
+  useEffect(() => {
+    socket.on("navigate", handleNavigate);
+    socket.on("host", handleHost);
+    socket.on("players:initial", handlePlayersInitial);
+    socket.on("players:add", handlePlayersAdd);
+    socket.on("players:remove", handlePlayersRemove);
+    socket.on("settings:initial", handleSettingsInitial);
+    socket.on("settings:update", handleSettingsUpdate);
+    socket.on("connect_error", handleConnectError);
+
+    return () => {
+      // Remove event handlers when component is unmounted to prevent buildup of identical handlers
+      socket.off("navigate", handleNavigate);
+      socket.off("host", handleHost);
+      socket.off("players:initial", handlePlayersInitial);
+      socket.off("players:add", handlePlayersAdd);
+      socket.off("players:remove", handlePlayersRemove);
+      socket.off("settings:initial", handleSettingsInitial);
+      socket.off("settings:update", handleSettingsUpdate);
+      socket.off("connect_error", handleConnectError);
+    };
+  });
```

Comment:
I'd add `[socket]` as a dependency to this hook 👍🏼 Tested it out locally & seems to work sweet